provider/provider.go

```diff
@@ -145,28 +207,48 @@ func (p *Provider) ConfigureProvider(_ context.Context, request *cqproto.Configu
 		}
 	}
 
+	var diags diag.Diagnostics
+
+	// validate provider configuration before proceeding
+	validateResponse, err := p.ValidateProviderConfig(ctx, &cqproto.ValidateProviderConfigRequest{Config: request.Config})
```

Comment:
Every call to `ConfigureProvider` also runs `ValidateProviderConfig`. This also means we decode and defaultize the config twice: once in validate and one down below.